frontend/src/scenes/insights/insightSceneLogic.tsx

```diff
@@ -132,17 +149,33 @@ export const insightSceneLogic = kea<insightSceneLogicType>([
             (s) => [
                 s.insightLogicRef,
                 s.insight,
+                s.dashboardId,
+                s.dashboardName,
                 groupsModel.selectors.aggregationLabel,
                 cohortsModel.selectors.cohortsById,
                 mathsLogic.selectors.mathDefinitions,
             ],
-            (insightLogicRef, insight, aggregationLabel, cohortsById, mathDefinitions): Breadcrumb[] => {
+            (
+                insightLogicRef,
+                insight,
+                dashboardId,
+                dashboardName,
+                aggregationLabel,
+                cohortsById,
+                mathDefinitions
+            ): Breadcrumb[] => {
                 return [
-                    {
-                        key: Scene.SavedInsights,
-                        name: 'Product analytics',
-                        path: urls.savedInsights(),
-                    },
+                    dashboardId !== null && dashboardName
+                        ? {
+                              key: Scene.Dashboard,
+                              name: dashboardName,
+                              path: urls.dashboard(dashboardId),
+                          }
```

Comment:
Only note is that the dashboard name should be preceded by "Dashboards", just like when you're on a dashboard. This will make for a smoother experience, as clicking into an insight will only append to the breadcrumbs

<img width="108" alt="Screenshot 2024-10-09 at 15 41 15" src="https://github.com/user-attachments/assets/dbf044a9-52f3-4d8d-b484-874d87946d7a">

Reply:
Great point, much cleaner now!